types/address.go

```diff
@@ -107,6 +111,11 @@ func init() {
 	}
 }
 
+// By default, caches are enabled. This enables or disables accAddrCache, consAddrCache, and valAddrCache.
+func SetAddrCacheEnabled(enabled bool) {
+	isCachingEnabled = enabled
+}
```

Comment:
It's unsafe to mutate a global variable like this without synchronization (e.g. a mutex).

Reply:
It's only meant to be used once on client startup. If I put a mutex in `SetAddrCacheEnabled` it should also be locked when checking `isCachingEnabled`, meaning every time an address bech32 conversion happens.

Reply:
You're correct that both reads and writes need to be synchronized. You might consider [atomic.Bool](https://pkg.go.dev/sync/atomic#Bool) as an alternative.

```go
var cachingEnabled atomic.Bool

func SetCachingEnabled(b bool) { cachingEnabled.Store(b) }
func GetCachingEnabled() bool { return cachingEnabled.Load() }
```